doc/architecture/decisions/0019-vulnerability-scanning.md

```diff
@@ -0,0 +1,33 @@
+# 19. Vulnerability scanning
+
+[Next >>](9999-end.md)
+
+Date: 2023-03-16
+
+## Status
+
+Draft for discussion
+
+## Context
+
+We currently use Trivy for our vulnerability scanning. It is used to scan all new dependencies added, through our pull
+requests, and also on a scheduled job for any new vulnerabilities found in our dependencies. We chose this method as it
+gave us a single point for all our vulnerability scanning.
+
+However, this approach does not scan all of our dependant containers, for example our xray sidecar, so we need a new
+approach.
+
+We also have now added in security hub into our AWS account, and will be using this as a centralised point for security notifications.
+
+## Approach
+
+As we now have Security Hub, we will now be using this as our central point for discovering new vulnerabilities in
+existing dependencies. All of our containers that we run, will now be gotten from our ECR, which will have continuous
+scanning. This continuous scanning will then output its findings into security hub, so we can see all the new
+developments there.
+
+For our new dependencies, we will still have Trivy scanning in our pull requests, as this will prevent the addition of
+new vulnerabilities.
```

Comment:
Consequence of this is that we need to enable Security Hub in all new accounts as and when they're enabled.

We may also need to enable cross account aggregation within Security Hub

Reply:
Another consequence here is that our container lifecycle policy may increase costs as it retains all images, which we should probably document.